metabase/metabase
# Browsing | Refactor navigation sidebar

The PR introduces a dedicated `MainNavbar` component instead of a combo of `CollectionSidebar` with some extra tweaks we currently have for the new navigation. The PR depends on some work from #21287 and #21288

This was a good opportunity to rethink how the collection sidebar tree used to work, so here are the most notable changes:

* completely switched to the `Tree` component that's used in the saved question picker. `Tree` has a lot of benefits (great work @alxnddr!):
   * a11y (HTML list markup, keyboard navigation support)
   * automatically expands selected item's parents when the tree mounts
   * scrolls to initially selected item on mount
* unified list item components used in the sidebar. Now bookmarks, collections, and links at the bottom are based on just two visual components sharing most of their code. As a result, it should be much easier to change their look consistently.
* fixed the selected element highlighting. Now when you open a bookmarked question/dashboard/model or open a link like "Browse data" or "Archive", the sidebar will highlight the selected element correctly
* fixed dragged collection item z-index. Previously, dragged element's z-index was lower than the sidebar's one, so the sidebar was overlapping it.

### To Verify

1. Open any page
2. Use the navigation sidebar and make sure nothing looks weird
